### ISSUE TITLE
Model 2 - LIGHTGBM

### ISSUE THREAD

**Issue by peperuizdev** [Bootcamp-IA-P4/data_scientist_g3#9]
# MODELO OPTIMIZADO 2 - LIGHTGBM

## 🎯 Objetivo
Implementar cuarto modelo optimizado con validación cruzada y tuning de hiperparámetros con Optuna. **Algoritmo específico se decidirá después del EDA.**

## 📅 Timeline
- **Inicio**: Miércoles 4/6 - 9:30
- **Deadline**: Jueves 9/6 - 9:00

### 1. **Preparación de Datos**
- [ ] Cargar datos procesados de `data/processed/`
- [ ] Configurar validación cruzada (5 folds)
- [ ] Aplicar técnicas de balanceo a variable objetivo (SMOTE) después de dividir datos en traint/test

### 2. **Optimización con Optuna**
- [ ] Definir función objetivo con métricas (F1 o AUC-ROC)
- [ ] Configurar espacio de búsqueda de hiperparámetros
- [ ] Ejecutar optimización con Optuna (50-100 trials)
- [ ] Analizar mejores parámetros encontrados

### 3. **Modelo Final**
- [ ] Entrenar modelo con mejores hiperparámetros
- [ ] Validación cruzada final
- [ ] Evaluar en test set con métricas completas

## 📊 Entregables
- [ ] `notebooks/modeling/model_[algoritmo].ipynb`
- [ ] Modelo optimizado guardado: `models/[algoritmo]_optimized.pkl`
- [ ] Métricas: Accuracy, Precision, Recall, F1, AUC-ROC
- [ ] Historial de optimización Optuna
- [ ] Análisis específico del modelo elegido
- [ ] Hiperparámetros finales documentados

## 🎯 Criterios de Éxito
- [ ] Optuna encuentra mejores hiperparámetros
- [ ] CV score promedio documentado
- [ ] Diferencia train/test <5%
- [ ] Modelo reproducible con seeds fijos
- [ ] Hiperparámetros justificados con Optuna

## 📝 Estructura del Notebook
```python
# 1. Cargar datos
# 2. Seleccionar todas las variables para entrenamiento y test.
# 3. Aplicar balanceo SMOTE o similares si procede
# 4. Entrenar modelo base para comparar con optimización
# 5. Definir función objetivo para Optuna
# 6. Configurar espacio de búsqueda de hiperparámetros
# 7. Ejecutar optimización con Optuna
# 8. Entrenar modelo final con mejores parámetros
# 9. Validación cruzada y evaluación en test set
# 10. Guardar modelo optimizado
